skeletons/NativeReal.c

```diff
@@ -361,15 +361,24 @@ NativeReal_decode_oer(const asn_codec_ctx_t *opt_codec_ctx,
          */
         NativeReal__network_swap(wire_size, ptr, scratch);
 
+
         switch(wire_size) {
-        case sizeof(double):
-            if(NativeReal__set(td, sptr, *(const double *)scratch) < 0)
-                ASN__DECODE_FAILED;
-            break;
-        case sizeof(float):
-            if(NativeReal__set(td, sptr, *(const float *)scratch) < 0)
-                ASN__DECODE_FAILED;
-            break;
+            case sizeof(double):
+                {
+                    double tmp;
+                    memcpy(&tmp, scratch, sizeof(double));
```

Comment:
That's one way to do it. Alternatively, the scratch could be a union of a buffer, double, float.

Do you have a particular reason to prefer this form?

Reply:
I read various stack overflow posts  :blush: and decided to play it safe
https://stackoverflow.com/questions/7630150/dereferencing-type-punned-pointer-will-break-strict-aliasing-rules#comment9265588_7630162
https://stackoverflow.com/questions/44245618/how-to-legally-use-type-punning-with-unions-to-cast-between-variations-of-struct?rq=1

I could try to rewrite it with a union if you prefer it this way

Reply:
No, as long as you picked it according to some reason, I am good with it.